### PR TITLE
chore(TUIN-143): dump github actions and migrate set-output

### PR DESCRIPTION
**Pull request by chrisnruud**

### CODE REVIEW

**Comment by chrisnruud** [comment]
OBS: Azure/aks-set-context er ikke oppgradert til v3 i denne commit siden det krever en hel del mer refaktorering for login - delen og muligens en download av kubelogin for at det skal fungere.
se https://github.com/Azure/aks-set-context

File: ci-cd/create-build-envs/action.yml [hunk 1]
```diff
@@ -384,13 +384,15 @@ runs:
         echo "${OUTPUT_ENV_JSON_NOT_SECRET}" | tee "${OUT_JSON_FILE}"
         echo "::endgroup::"
 
-        echo "::set-output name=json-without-secrets-path::${OUT_JSON_FILE}"
-        echo "::set-output name=build-envs-artifact-name::build-envs-$(get-val 'application-version')"
+        echo 'json-without-secrets-path<<"${{ github.run_id }}"' >> $GITHUB_OUTPUT
+        echo "${OUT_JSON_FILE}" >> $GITHUB_OUTPUT
```
Comment (Laffs2k5) [inline ci-cd/create-build-envs/action.yml:388]: Denne er ikke multiline ;-)

File: .github/workflows/update-deps-and-create-pr.yml [hunk 2]
```diff
@@ -84,7 +84,7 @@ jobs:
       - name: 🌱 Commit to new branch (or update existing) & create pull request
         # Will update branch if it already exists.
         # See https://github.com/peter-evans/create-pull-request for details.
-        uses: peter-evans/create-pull-request@bd72e1b7922d417764d27d30768117ad7da78a0e # 4.0.2
+        uses: peter-evans/create-pull-request@b4d51739f96fca8047ad065eccef63442d8e99f7 # 4.2.0
```
Comment (Laffs2k5) [inline .github/workflows/update-deps-and-create-pr.yml:87]: Was the diff reviewed before pinning?

**Comment by Laffs2k5** [comment]
> OBS: Azure/aks-set-context er ikke oppgradert til v3 i denne commit siden det krever en hel del mer refaktorering for login - delen og muligens en download av kubelogin for at det skal fungere. se https://github.com/Azure/aks-set-context

You should be good using `v3` by moving the creds to a separate login action call before `aks-set-context` like this:
````yaml
- uses: azure/login@v1
  with:
     creds: ${{ fromJSON(inputs.dsb-build-envs).pr-deploy-aks-creds }}
- uses: azure/aks-set-context@v3
  with:
    cluster-name: ${{ fromJSON(inputs.dsb-build-envs).pr-deploy-aks-cluster-name }}
    resource-group: ${{ fromJSON(inputs.dsb-build-envs).pr-deploy-aks-resource-group }}
```

Applies to:
- ci-cd/teardown-pr-environment/action.yml
- ci-cd/deploy-to-ephemeral/action.yml

**Review by Laffs2k5 — APPROVED** [review]
Dont forget the part about releasing after merge to main:
https://github.com/dsb-norge/github-actions/blob/main/ci-cd/README.md#release

As well as removing the temporary `tuin-143` tag ;-)

File: .github/workflows/update-deps-and-create-pr.yml (hunk 2 above)
Comment (chrisnruud) [reply .github/workflows/update-deps-and-create-pr.yml:87]: I did a cursory review of the release

**Comment by chrisnruud** [comment]
> > OBS: Azure/aks-set-context er ikke oppgradert til v3 i denne commit siden det krever en hel del mer refaktorering for login - delen og muligens en download av kubelogin for at det skal fungere. se https://github.com/Azure/aks-set-context
> 
> You should be good using `v3` by moving the creds to a separate login action call before `aks-set-context` like this:
> ```yaml
> - uses: azure/login@v1
>   with:
>      creds: ${{ fromJSON(inputs.dsb-build-envs).pr-deploy-aks-creds }}
> - uses: azure/aks-set-context@v3
>   with:
>     cluster-name: ${{ fromJSON(inputs.dsb-build-envs).pr-deploy-aks-cluster-name }}
>     resource-group: ${{ fromJSON(inputs.dsb-build-envs).pr-deploy-aks-resource-group }}
> ```
> 
> Applies to:
> - `ci-cd/teardown-pr-environment/action.yml`
> - `ci-cd/deploy-to-ephemeral/action.yml`

Prøvde dette her:
https://github.com/dsb-norge/eksplosiv-org/actions/runs/3272455676/jobs/5383703261
Og da ble det mas om kubelogin så jeg tar det for seg.

**Pull request merged**